src/devices/cellular.c

```diff
@@ -419,9 +441,11 @@ int cellular_init_connection(DeviceConfig *config, millis_t * connected_at)
         /* This is sane since DeviceConfig is typedef'd as a serial_buffer */
         struct serial_buffer *sb = (struct serial_buffer*) config;
 
-        const int init_status = cellular_init_modem(sb);
-        if (DEVICE_INIT_SUCCESS != init_status)
-                return DEVICE_INIT_FAIL;
+        if (hard_init) {
+                const int init_status = cellular_init_modem(sb);
```

Comment:
So we only initialize now if hard init is set?  We otherwise don't init?  Seems off.

Reply:
We pass in a flag for hard initialization - this is controlled by the outside routine (connectivity task). 

If the hard_init is false, we bypass the power cycle routine, which will speed up cellular acquisition. 

The gist is, if we lose connection, we give the system a chance to quickly re-connect. 

The logic in the connectivity task will do a 'fast init' (i.e. hard_init = false) up to X amount of times before it falls back to power cycling.